src/cli/cli.cpp

```diff
@@ -3896,5 +3896,47 @@ Interpreter &Interpreter::GetOwner(OwnerLocator &aOwnerLocator)
     return interpreter;
 }
 
+extern "C" void otCliSetUserCommands(const otCliCommand *aUserCommands, uint8_t aLength)
+{
+    Server::sServer->GetInterpreter().SetUserCommands(aUserCommands, aLength);
+}
+
+extern "C" void otCliOutputBytes(const uint8_t *aBytes, uint8_t aLength)
+{
+    Server::sServer->GetInterpreter().OutputBytes(aBytes, aLength);
+}
+
+extern "C" void otCliOutputFormat(const char *aFmt, ...)
+{
+    va_list aAp;
+    va_start(aAp, aFmt);
+    static_cast<Uart *>(Server::sServer)->OutputFormatV(aFmt, aAp);
```

Comment:
This function still uses UART specific `OutputFormatV` method.

Actually, `OutputFormatV` is quite generic as it only calls virtual `Output` method. Perhaps it could become a part of the base class?

Reply:
I didn't notice that this work is already done in paralel PR #3502. I think then we should wait for #3502 to be merged, and then rebase.

Reply:
#3502 is merged now. @bukepo, can you update this PR?